common/src/rv_trace.rs

```diff
@@ -355,6 +330,28 @@ impl ELFInstruction {
             | RV32IM::BGEU,
         );
 
+        // TODO(moodlezoup): Use these flags in R1CS constraints
+        flags[9] = match self.virtual_sequence_index {
+            // For virtual sequences, we set
+            //     virtual PC := ProgARW     (the bytecode `a` value)
+            // if it's the first instruction in the sequence.
+            // Otherwise, we increment the virtual PC:
+            //     virtual PC += 1
+            // This prevents a malicious prover from reordering or omitting
+            // instructions from the virtual sequence.
+            Some(i) => i > 0,
+            // For "real" instructions, we always set
+            //     virtual PC := ProgARW     (the bytecode `a` value)
+            None => false
```

Comment:
@arasuarun does this sound right to you? (I haven't actually updated the constraints yet)